src/k8s-engine/lima.ts

```diff
@@ -726,33 +726,19 @@ export default class LimaBackend extends events.EventEmitter implements K8s.Kube
       return;
     }
     await this.showSudoReason(explanations);
+    const singleCommand = commands.join('; ');
 
-    const tmpScript = path.join(os.tmpdir(), `rd-sudo-commands-${ randomTag }.sh`);
-    const logFile = path.join(os.tmpdir(), `rd-sudo-commands-run-${ randomTag }.log`);
-
-    await fs.promises.writeFile(tmpScript, `#!/usr/bin/env bash
-
-exec &> >(tee ${ logFile })
-set -ex
-
-${ commands.join('\n') }
-`,
-    { mode: 0o700 });
+    if (singleCommand.includes("'")) {
+      throw new Error(`Can't execute commands ${ singleCommand } because there's a single-quote in them.`);
+    }
     try {
-      await this.sudoExec(tmpScript);
+      await this.sudoExec(`/bin/sh -c '${ singleCommand }'`);
```

Comment:
We're switching from `/usr/bin/env bash` to `/bin/sh` here; is that okay?

Reply:
I think so. Anything you run via `sudo` should not rely on the environment to look up executables.

The commands should all be basic POSIX shell commands. At least in our Alpine image `bash` is just a wrapper around `/bin/sh`:

```
$ rdlima cat /bin/bash
#!/bin/sh
exec /bin/ash "$@"
```